### ISSUE TITLE
fetch_open_data fails if NaNs are present

### ISSUE THREAD

**Issue by GCArullo** [gwpy/gwpy#1126]
When the data surrounding an event contain NaNs (which I think is the standard characterization for non-science data),  gwpy.timeseries.TimeSeries.fetch_open_data fails.
I am using python 2.7, gwpy 0.12.2 and the OS is macOS High Sierra 10.13.4.

Example using 4096s surrounding GW170729:

```python
from gwpy.timeseries import TimeSeries
data = TimeSeries.fetch_open_data('L1', 1185387759, 1185391855)
```

error:

```pytb
>>> data = TimeSeries.fetch_open_data('L1', 1185387759, 1185391855)
  File "/Users/gregoriocarullo/anaconda2/lib/python2.7/site-packages/gwpy-0.12.2-py2.7.egg/gwpy/timeseries/core.py", line 454, in fetch_open_data
    host=host, cls=cls, **kwargs)
  File "/Users/gregoriocarullo/anaconda2/lib/python2.7/site-packages/gwpy-0.12.2-py2.7.egg/gwpy/timeseries/io/losc.py", line 289, in fetch_losc_data
    format=format)
  File "/Users/gregoriocarullo/anaconda2/lib/python2.7/site-packages/gwpy-0.12.2-py2.7.egg/gwpy/timeseries/io/losc.py", line 202, in find_losc_urls
    start, end, tag=tag, version=version)
  File "/Users/gregoriocarullo/anaconda2/lib/python2.7/site-packages/gwpy-0.12.2-py2.7.egg/gwpy/timeseries/io/losc.py", line 92, in _match_urls
    m = _match_url(url, start, end, tag=tag, version=version)
  File "/Users/gregoriocarullo/anaconda2/lib/python2.7/site-packages/gwpy-0.12.2-py2.7.egg/gwpy/timeseries/io/losc.py", line 133, in _match_url
    reg = LOSC_URL_RE.match(os.path.basename(url)).groupdict()
AttributeError: 'NoneType' object has no attribute 'groupdict'
```

**Comment by duncanmmacleod** [comment]
@GCArullo, thanks for the report. I'm confused with your description of the problem, the traceback you show indicates the gwpy is failing to parse the file names, and hasn't actually gotten to the point of trying to read the data.

However, I also see that you are using `gwpy-0.12.2`, can you try and reproduce the issue with `gwpy>=0.15.0`?

**Comment by duncanmmacleod** [comment]
No response, so will close, please reopen if this is still an issue.

**Comment by GCArullo** [comment]
Sorry for the delayed reply. The issue is indeed solved when using gwpy>=0.15.0. Thanks for the help.
